libs/wingc/src/jsify.rs

```diff
@@ -304,7 +310,14 @@ impl<'a> JSifier<'a> {
 						.iter()
 						.map(|p| match p {
 							InterpolatedStringPart::Static(l) => format!("{}", l),
-							InterpolatedStringPart::Expr(e) => format!("${{{}}}", self.jsify_expression(e, phase)),
+							InterpolatedStringPart::Expr(e) => {
+								match *e.evaluated_type.borrow().expect("Should have type") {
+									Type::Json | Type::MutJson => {
+										format!("${{JSON.stringify({}, null, 2)}}", self.jsify_expression(e, phase))
+									}
+									_ => format!("${{{}}}", self.jsify_expression(e, phase)),
```

Comment:
@eladb - this is the line of code that added support for `bool` in string interpolation (and actually any expression), which is why @hasanaburayyan deleted `examples/tests/invalid/types_string_interpolation.w`

Reply:
Ah yea I meant to start a conversation about why we were not allowing prior. 😅